src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddDebugLogProcessorPass.php

```diff
@@ -22,22 +23,38 @@ public function process(ContainerBuilder $container)
         if (!$container->hasDefinition('profiler')) {
             return;
         }
-        if (!$container->hasDefinition('monolog.logger_prototype')) {
+
+        if ($container->hasDefinition('monolog.logger_prototype') && $container->hasDefinition('debug.log_processor')) {
+            $container->getDefinition('monolog.logger_prototype')
+                ->setConfigurator([__CLASS__, 'configureMonologLogger'])
+                ->addMethodCall('pushProcessor', [new Reference('debug.log_processor')])
+            ;
+
             return;
         }
-        if (!$container->hasDefinition('debug.log_processor')) {
+
+        if (!$container->hasDefinition('logger')) {
             return;
         }
 
-        $definition = $container->getDefinition('monolog.logger_prototype');
-        $definition->setConfigurator([__CLASS__, 'configureLogger']);
-        $definition->addMethodCall('pushProcessor', [new Reference('debug.log_processor')]);
+        $loggerDefinition = $container->getDefinition('logger');
+
+        if (Logger::class === $loggerDefinition->getClass()) {
+            $loggerDefinition->setConfigurator([__CLASS__, 'configureHttpKernelLogger']);
+        }
     }
 
-    public static function configureLogger(mixed $logger)
+    public static function configureMonologLogger(mixed $logger)
     {
-        if (\is_object($logger) && method_exists($logger, 'removeDebugLogger') && \in_array(\PHP_SAPI, ['cli', 'phpdbg'], true)) {
+        if (\in_array(\PHP_SAPI, ['cli', 'phpdbg'], true) && \is_object($logger) && method_exists($logger, 'removeDebugLogger')) {
             $logger->removeDebugLogger();
         }
     }
+
+    public static function configureHttpKernelLogger(Logger $logger)
+    {
+        if (method_exists($logger, 'enableDebug')) {
+            $logger->enableDebug(!\in_array(\PHP_SAPI, ['cli', 'phpdbg'], true));
```

Comment:
the check was better first in the if, opcache again

Reply:
But `Logger::$debug` is `false` by default so I need a way to make it `true` if needed :thinking: 

Seems it conflicts with https://github.com/symfony/symfony/pull/47483#discussion_r997270971

Reply:
I'm likely missing something yes :)
why wouldn't this work?
```php
        if (!\in_array(\PHP_SAPI, ['cli', 'phpdbg'], true) && method_exists($logger, 'enableDebug')) {
            $logger->enableDebug(true);
        }
```